pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -821,6 +822,66 @@ func (r *ReconcileHyperConverged) ensureKubeVirtTemplateValidator(instance *hcov
 	return r.client.Status().Update(context.TODO(), instance)
 }
 
+func newKubeVirtStorageConfigForCR(cr *hcov1alpha1.HyperConverged, namespace string) *corev1.ConfigMap {
+	var volumeMode string
+
+	if *(&cr.Spec.BareMetalPlatform) {
+		volumeMode = "Block"
+	} else {
+		volumeMode = "Filesystem"
+	}
+
+	labels := map[string]string{
+		"app": cr.Name,
+	}
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubevirt-config-storage-class-defaults",
+			Labels:    labels,
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			"accessMode":          "ReadWriteMany",
+			"volumeMode":          volumeMode,
+			"local-sc.accessMode": "ReadWriteOnce",
```

Comment:
@danielerez , this assumes that the target cluster has a storage class called `local-sc`. It's up to cluster's admin how (s)he calls it and what particular SC is behind that name.

We can either
- autodetect
- or provide a way for manual intervention

Reply:
The admin can change it by modifying the ConfigMap (reconcile won't change it once already created). @rthallisey do we want to add the local class name to HyperConvergedSpec as well or is it too specific?

Reply:
@mareklibra @rthallisey Added 'LocalStorageClassName' key to HyperConverged spec, so it could be configured on deployment.